src/vtk.cpp

```diff
@@ -119,24 +222,44 @@ std::vector<std::string> split (const std::string &s, char delim) {
   std::stringstream ss (s);
   std::string item;
 
-  while (getline (ss, item, delim)) {
-      result.push_back (item);
+  while (getline(ss, item, delim)) {
+    result.push_back(item);
   }
 
   return result;
 }
 
-void VtkOutput::write(const Hydro h, std::string &quantity){
-  std::vector<std::string> quantity_list=split(quantity,',');
-  vtk_output_counter_++;
-  for (auto q : quantity_list){
-    std::string t=q;
+bool VtkOutput::is_quantity_implemented(std::string &quantity) {
+  bool quantity_is_valid = (valid_quantities_.find(quantity)
+                            != valid_quantities_.end());
+  return quantity_is_valid;
+}
+
+void VtkOutput::write(const Hydro h, std::string &quantities) {
+  std::vector<std::string> quantities_list = split(quantities,',');
+  for (std::string q : quantities_list){
+    if (!is_quantity_implemented(q)) {
+      std::cout << "Given quantity '" << q << "' is not an "
+        "implemented VTK quantity. This entry will be skipped." << std::endl;
+      continue;
+    }
     std::ofstream file;
 
-    file.open(make_filename(t, vtk_output_counter_), std::ios::out);
-    write_header(file, h, t);
-    write_vtk_scalar(file, h, q);
+    file.open(make_filename(q, vtk_output_counter_), std::ios::out);
+    write_header(file, h, q);
+    if (valid_quantities_.at(q) == "scalar") {
+      write_vtk_scalar(file, h, q);
+    } else if (valid_quantities_.at(q) == "vector") {
+      write_vtk_vector(file, h, q);
+    } else if (valid_quantities_.at(q) == "tensor") {
+      write_vtk_tensor(file, h, q);
+    } else {
+      std::cout << "Quantity '" << q << "' is neither stated to be a scalar, "
+        "nor a vector, nor a tensor. Skipping this quantity. Please check the "
+        "map in file src/vtk.h." << std::endl;
+    }
   }
-  
+
```

Comment:
There should be probably also a file close here.

Reply:
True, done.